### ISSUE TITLE
Release 4.8.0

### ISSUE THREAD

**Issue by blairconrad** [FakeItEasy/FakeItEasy#1392]
Can be labelled **ready** when all other issues on this milestone are closed.

- [x] Address all coding and style violations detected during a *Release* build (send a regular pull request which must be merged before continuing)
- [x] ensure completeness of draft [release notes](https://github.com/FakeItEasy/FakeItEasy/releases), including non-owner contributors, if any (move release notes forward from any pre-releases to the current release)
- [x] run `tools\prepare_release.cmd x.y.z` to:
    - change "vNext" names to appropriate x.y.z version for the next GitHub Release, milestone, and release checklist (this issue)
    - create a new draft GitHub Release, milestone, and release checklist
    - create release/x.y.z branch on upstream, branching from develop
    - submit a pull request to merge the release/x.y.z branch to master
- [x] merge the pull request, _which may be done by the PR creator, unlike a regular pull request_
- [x] tag master with the new version number and push to upstream. This will cause the build server to
    - push the nupkgs to [nuget.org](https://www.nuget.org/profiles/FakeItEasy) and
    - attach all nupkgs from the tagged build to the GitHub Release.
- [x] wait for NuGet package validation
- [x] de-list pre-release NuGet packages if present
- [x] publish the GitHub release
- [x] tweet, mentioning contributors, and post link as comment here for easy retweeting ;-)
- [x] post tweet in [Gitter](https://gitter.im/FakeItEasy/FakeItEasy)
- [x] post a link to the GitHub Release in each issue in this milestone, with thanks to contributors
- [x] close this issue
- [ ] close the [milestone](https://github.com/FakeItEasy/FakeItEasy/milestones?state=open) for this release

**Comment by blairconrad** [comment]
@thomaslevesque, we've 4 fixes and an enhancement. Time for a release? I can do on Wednesday morning.

**Comment by blairconrad** [comment]
Oh, maybe we should squeeze #1402 in first.

**Comment by thomaslevesque** [comment]
Sure, let's go!

**Comment by thomaslevesque** [comment]
BTW, https://github.com/FakeItEasy/FakeItEasy/issues/1353 is technically a breaking change. Previously, specifying custom attributes or constructor arguments for a fake delegate had no effect, now it will throw an exception. Should have thought of this before merging...

**Comment by blairconrad** [comment]
Sorry, yes. I was aware, but in my head these sorts of bugs where the API lets you think you're doing something and it's not having the effect you want warrant the break (and I doubt it'll affect anyone). If you are concerned, though, we can roll back and release without it.

**Comment by thomaslevesque** [comment]
Oh, I agree with you. I just thought it was worth mentioning. We should include it.

**Comment by blairconrad** [comment]
Cool, thanks. I mentioned the potential to break a test in the release notes.

**Comment by blairconrad** [comment]
Proposed tweet:

```
Get FakeItEasy 4.8.0!
- Failure to create Dummy throws correct exception and now includes reasons for failure.
- Use of illegal delegate-faking options throws immediately.
- Fake creation no longer trying parameterless constructor when there isn't one.

https://github.com/FakeItEasy/FakeItEasy/releases/4.8.0
```

**Comment by thomaslevesque** [comment]
👍

**Comment by blairconrad** [comment]
https://twitter.com/fakeiteasyfx/status/1027145925406404608

**Comment by blairconrad** [comment]
This change has been released as part of [FakeItEasy 4.8.0](https://github.com/FakeItEasy/FakeItEasy/releases/4.8.0).

**Comment by blairconrad** [comment]
Thanks, @thomaslevesque! o/
